Test suite failure with GHC 8

```
> /tmp/stackage-build8/multiset-0.3.2$ runghc -clear-package-db -global-package-db -package-db=/home/stackage/work/builds/nightly/pkgdb Setup configure --enable-tests --package-db=clear --package-db=global --package-db=/home/stackage/work/builds/nightly/pkgdb --libdir=/home/stackage/work/builds/nightly/lib --bindir=/home/stackage/work/builds/nightly/bin --datadir=/home/stackage/work/builds/nightly/share --libexecdir=/home/stackage/work/builds/nightly/libexec --sysconfdir=/home/stackage/work/builds/nightly/etc --docdir=/home/stackage/work/builds/nightly/doc/multiset-0.3.2 --htmldir=/home/stackage/work/builds/nightly/doc/multiset-0.3.2 --haddockdir=/home/stackage/work/builds/nightly/doc/multiset-0.3.2 --flags=
Configuring multiset-0.3.2...
> /tmp/stackage-build8/multiset-0.3.2$ runghc -clear-package-db -global-package-db -package-db=/home/stackage/work/builds/nightly/pkgdb Setup build
Building multiset-0.3.2...
Preprocessing library multiset-0.3.2...
[1 of 2] Compiling Data.MultiSet    ( Data/MultiSet.hs, dist/build/Data/MultiSet.o )

Data/MultiSet.hs:139:1: warning: [-Wdodgy-imports]
    Module ‘Prelude’ does not export ‘join’

Data/MultiSet.hs:419:1: warning: [-Wredundant-constraints]
    • Redundant constraint: Ord a
    • In the type signature for:
           filter :: Ord a => (a -> Bool) -> MultiSet a -> MultiSet a

Data/MultiSet.hs:425:1: warning: [-Wredundant-constraints]
    • Redundant constraint: Ord a
    • In the type signature for:
           partition :: Ord a =>
                        (a -> Bool) -> MultiSet a -> (MultiSet a, MultiSet a)

Data/MultiSet.hs:580:1: warning: [-Wredundant-constraints]
    • Redundant constraint: Ord a
    • In the type signature for:
           fromMap :: Ord a => Map a Occur -> MultiSet a
[2 of 2] Compiling Data.IntMultiSet ( Data/IntMultiSet.hs, dist/build/Data/IntMultiSet.o )

Data/IntMultiSet.hs:133:1: warning: [-Wdodgy-imports]
    Module ‘Prelude’ does not export ‘join’
Preprocessing test suite 'doctests' for multiset-0.3.2...
[1 of 1] Compiling Main             ( test/Main.hs, dist/build/doctests/doctests-tmp/Main.o )
Linking dist/build/doctests/doctests ...
> /tmp/stackage-build8/multiset-0.3.2$ dist/build/doctests/doctests

/tmp/stackage-build8/multiset-0.3.2/Data/IntMultiSet.hs:671:0: error:
     fatal error: Typeable.h: No such file or directory
     #include "Typeable.h"
     ^
compilation terminated.
doctests: `gcc' failed in phase `C pre-processor'. (Exit code: 1)
```

--- a/Data/IntMultiSet.hs
+++ b/Data/IntMultiSet.hs
@@ -672,8 +672,10 @@ instance Read IntMultiSet where
   Typeable/Data
 --------------------------------------------------------------------}
 
+#if __GLASGOW_HASKELL__ < 800
 #include "Typeable.h"
 INSTANCE_TYPEABLE0(IntMultiSet,intMultiSetTc,"IntMultiSet")
+#endif
 
 {--------------------------------------------------------------------
   Split

--- a/Data/MultiSet.hs
+++ b/Data/MultiSet.hs
@@ -666,8 +666,10 @@ instance (Read a, Ord a) => Read (MultiSet a) where
   Typeable/Data
 --------------------------------------------------------------------}
 
+#if __GLASGOW_HASKELL__ < 800
 #include "Typeable.h"
 INSTANCE_TYPEABLE1(MultiSet,multiSetTc,"MultiSet")
+#endif
 
 {--------------------------------------------------------------------
   Split

